tests/command_line/check_main.py

```diff
@@ -51,13 +53,105 @@ def check_symlink(self):
         os.symlink(old_results, self.latest_symlink)
         assert os.path.islink(self.latest_symlink) and os.path.exists(self.latest_symlink)
 
-        setup_results_directory(self.results_root, self.results_dir)
+        setup_results_directory(self.results_dir)
         update_latest_symlink(self.results_root, self.results_dir)
         self.validate_directories()
 
         # Try again if symlink exists and points to nothing
         os.rmdir(self.results_dir)
         assert os.path.islink(self.latest_symlink) and not os.path.exists(self.latest_symlink)
-        setup_results_directory(self.results_root, self.results_dir)
+        setup_results_directory(self.results_dir)
         update_latest_symlink(self.results_root, self.results_dir)
         self.validate_directories()
+
+
+globals_json = """
+{
+    "x": 200
+}
+"""
+
+invalid_globals_json = """
+{
+    can't parse this!: ?right?
+}
+"""
+
+valid_json_not_dict = """
+[
+    {
+        "x": 200,
+        "y": 300
+    }
+]
+"""
+
+
+class CheckUserDefinedGlobals(object):
+    """Tests for the helper method which parses in user defined globals option"""
+
+    def check_immutable(self):
+        """Expect the user defined dict object to be immutable."""
+        global_dict = get_user_defined_globals(globals_json)
+        try:
+            global_dict["x"] = -1
+            global_dict["y"] = 3
+            assert False, "globals dict should be immutable"
+        except:
+            # expected
+            pass
+
+    def check_parseable_json_string(self):
+        globals_dict = get_user_defined_globals(globals_json)
+        assert globals_dict == json.loads(globals_json)
+
+    def check_unparseable(self):
+        """Check for expected error"""
+        try:
+            global_dict = get_user_defined_globals(invalid_globals_json)
+            assert False, "Should have failed to parse invalid JSON"
+        except:
```

Comment:
It's preferable to expect a specific type of exception. See https://pytest.org/latest/assert.html#assertions-about-expected-exceptions Most of these tests could use the same treatment